src/js/index.js

```diff
@@ -133,16 +149,87 @@ document.addEventListener("DOMContentLoaded", function () {
   init();
 });
 
-// 1. 브라우져 렌더링 과정
-// dom tree cssom tree render tree
-// script tag (defer async)
 
-// 2. DOMContentLoaded, window.onload 돔이 렌더링이 된 이후에 울리는 함수 (브라우져)
+// delegate 함수 구현 (중간중간 console.log) 고도화
+
+// 유사배열, 배열
+
+// 아래 함수 구현
+
+// 생활코딩 git 수업
+
+// 페이지를 로드
+function loadMenu() {
+  localStorage.setItem('espresso', JSON.stringify({
+      espresso: [{
+          id: 1,
+          name: 'papico'
+      }, {
+          id: 2,
+          name: 'siwon'
+      }]
+  }));
 
-// 3. querySelectorAll, querySelector
+  loging();
+}
+
+loadMenu();
+
+
+// 메뉴추가
+function addMenu() {
+  const espressoMenuList = JSON.parse(localStorage.getItem('espresso'));
+  
+  const newMenu = {
+      id: 3,
+      name: 'cool-jung'
+  };
+  const copyEspressoMenuList = [...espressoMenuList.espresso,newMenu];
+  localStorageSetItem({espresso,copyEspressoMenuList});
+}
+addMenu();
+
+// 메뉴 삭제
+function removeMenu() {
+  const espressoMenuList = JSON.parse(localStorage.getItem('espresso'));
+  console.log(espressoMenuList.espresso);
+  const deleteMenuId = 1;
+  const removeMenuList = espressoMenuList.espresso.filter((element)=>{
+      return element.id !== deleteMenuId;
+  });
+  console.log(removeMenuList);
+  
+  localStorageSetItem({espresso,removeMenuList});
+}
+removeMenu();
+
+// 메뉴 수정
+function editMenu(){
+  const espressoMenuList = JSON.parse(localStorage.getItem('espresso'));
+  const editMenu = {
+      id: 2,
+      name: 'luya'
+  };
+  const updatedName = espressoMenuList.espresso.map((element)=>{
+      if(element.id == 2){
+          element.name = editMenu.name;
+          return element;
+      }else{
+          return element;
+      }
+  });
+  console.log(updatedName);
+  localStorageSetItem({espresso,updatedName});
+}
+editMenu();
 
-// 4. delegate 함수 구현 (중간중간 console.log)
+// 로깅
+function loging(){
+  const espressoMenuList = JSON.parse(localStorage.getItem('espresso'));
 
-// 5. 유사배열, 배열
+  console.log(espressoMenuList);
+}
 
-// 6. 삭제를 delegate 구현하기
\ No newline at end of file
+function localStorageSetItem({key,val}){
+  localStorage.setItem('espresso',JSON.stringify({key:val}));
```

Comment:
```suggestion
  localStorage.setItem('espresso',JSON.stringify({[key]:val}));
```
넘겨받은 변수를 객체의 키로 사용하기 위해서는 `[]` 로 묶어줘야 합니다.